### ISSUE TITLE
Suggestions of how to improve this needed

### ISSUE THREAD

**Issue by ThorvaldAagaard** [lorserker/ben#66]
In the following position:
![image](https://github.com/lorserker/ben/assets/10560652/56b82586-f5d9-43b0-97ac-6a6e089a3bd6)

the suggested card from the neural network is just a bad play.

**Comment by lorserker** [comment]
i agree. it's better to test the spades first. if they don't split, we still have the option of a squeeze. the squeeze is clearly superior if East has both spades. If West has the two spades then both the squeeze and the finesse are an option.
Now Ben has committed itself to the finesse before knowing what is going on.

I think that there are several problems
- the NN is not advanced enough to understand cardplay technique described above
- the DD solver believes that playing D is exactly as good as playing SA, because it will only make the finesse when it actually works and will try for the spades to split otherwise. in the samples there is no case where a squeeze against West would be the correct play.

a relevant question: did West show out of spades earlier in the play?

**Comment by lorserker** [comment]
yes, as it is, the NN is just too attracted to play towards the AQ because it has seen that done millions of times probably :)
the NN doesn't try to play well, it tries to play similar to what is in the data.
the NN doesn't "see" the spade finesse because it sees the 7 as x.

**Comment by ThorvaldAagaard** [comment]
Newwest version will bid to 4S, but there are some bad play from PIMC, so keeping this open for now

[Event "?"]
[Site "?"]
[Date "?"]
[Board "69"]
[West "?"]
[North "?"]
[East "?"]
[South "?"]
[Dealer "S"]
[Vulnerable "EW"]
[Deal "N:T982.732.K87.J98 J53.96.AQ6542.64 K4.QJ854.JT9.732 AQ76.AKT.3.AKQT5"]
[Scoring "?"]
[Declarer "?"]
[Contract "?"]
[Result "?"]
[Info "None"]

![Image](https://github.com/user-attachments/assets/ce669156-a34d-454b-80e1-2753d4c64eb0)

**Comment by ThorvaldAagaard** [comment]
Same family - committing to the finesse before testing spades. In #188. Closing.
